src/java/org/apache/cassandra/index/SecondaryIndexManager.java

```diff
@@ -490,10 +490,58 @@ public static String getIndexName(String cfName)
         assert isIndexColumnFamily(cfName);
         return StringUtils.substringAfter(cfName, Directories.SECONDARY_INDEX_NAME_SEPARATOR);
     }
+    
+    public void buildIndexesBlockingIncremental(Collection<SSTableReader> sstables)
```

Comment:
I'll JavaDoc this when we're closer having the patch approved, but basically it's a version of `buildIndexesBlocking()` that doesn't touch the index queryability/status. It is here to build indexes for new SSTables before they are actually added to the tracker, and so failures have no bearing on those things...except, of course, for failing the upstream operation and making sure the SSTables themselves never become readable.

Reply:
Open to any suggestions on the logging, structure, possibilities for deduplication, etc.

Reply:
Added JavaDoc.